src/components/UserSupervisorEditable/UserSupervisorEditable.tsx

```diff
@@ -28,21 +30,25 @@ export const UserSupervisorEditable = ({ onChange }: UserSupervisorEditableProps
             onChange={(value: User) => {
                 setSearch(value.name || value.email);
                 setSelectedSupervisor(value);
+                setComboboxVisibility(false);
                 onChange(value);
             }}
-            visible={!selectedSupervisor}
+            visible={comboboxVisibility}
             items={userListQuery.data}
-            maxWidth={100}
             renderInput={({ value, ...restProps }) => (
                 <Input
+                    iconLeft={nullable(selectedSupervisor, (s) => (
+                        <UserPic size={18} name={s.name} src={s.image} email={s.email} />
+                    ))}
                     placeholder={tr('Choose a supervisor')}
                     autoFocus
                     size="m"
                     autoComplete="off"
-                    value={value ?? search}
+                    value={value ?? selectedSupervisor?.name}
                     onChange={(e) => {
-                        setSelectedSupervisor(undefined);
+                        setSelectedSupervisor(null);
```

Comment:
Why do we need set supervisor null in input onChange?

Reply:
You have a selected user
You start to erase the name
Input reacts by unassigning current user, since the name doesn't match anymore